### DISCUSSION TITLE
"non-existing domain" error - car unavailable

### DISCUSSION THREAD

**Discussion by lits101** [teslamate-org/teslamate#3779] [Q&A]
On 15/3 my local docker setup of Teslamate 1.28.3 stopped talking to my car and was asking for my tokens again, but told me they were invalid when supplied. 

I upgraded to 1.28.5 today and I can get into Teslamate again without having to provide my tokens, but the car shows up as unavailable. Logs show issues reaching domains including owner-api.tesla.com and api.github.com. 

The below issue from Discussions shows someone else having the same issue. 
https://github.com/teslamate-org/teslamate/discussions/3213#discussioncomment-8858750

> I am having an extremely similar issue on a homeserver,  the output of the log is.
> 
> I am able to ping google as well as the auth.tesla.com and I just updated my docker to the current version, any ideas?
> 
> `{"log":"2024-03-20 22:31:24.918 [info] GET /\n","stream":"stdout","time":"2024-03-20T22:31:24.919136293Z"}
> 
> {"log":"2024-03-20 22:31:24.920 [info] Sent 302 in 1ms\n","stream":"stdout","time":"2024-03-20T22:31:24.920428236Z"}
> 
> {"log":"2024-03-20 22:31:24.925 [info] GET /sign_in\n","stream":"stdout","time":"2024-03-20T22:31:24.925839243Z"}
> 
> {"log":"2024-03-20 22:31:24.927 [info] Sent 200 in 2ms\n","stream":"stdout","time":"2024-03-20T22:31:24.927325186Z"}
> 
> {"log":"2024-03-20 22:31:43.029 [error] POST https://auth.tesla.com/oauth2/v3/token -\u003e error: \"non-existing domain\" (8001.748 ms)\n","stream":"stdout","time":"2024-03-20T22:31:43.>
> 
> {"log":"2024-03-20 22:32:17.501 [error] GET https://api.github.com/repos/teslamate-org/teslamate/releases/latest -\u003e error: \"non-existing domain\" (8000.923 ms)\n","stream":"stdout">
> 
> {"log":"2024-03-20 22:32:17.501 [warning] Update check failed: \"non-existing domain\"\n","stream":"stdout","time":"2024-03-20T22:32:17.502121942Z"}
> 
> {"log":"2024-03-20 22:34:39.499 [warning] Cannot refresh access token: :not_signed_in\n","stream":"stdout","time":"2024-03-20T22:34:39.499775561Z"}
> 
> {"log":"2024-03-20 22:41:28.428 [info] GET /\n","stream":"stdout","time":"2024-03-20T22:41:28.429206233Z"}
> 
> {"log":"2024-03-20 22:41:28.430 [info] Sent 302 in 1ms\n","stream":"stdout","time":"2024-03-20T22:41:28.430346127Z"}
> 
> {"log":"2024-03-20 22:41:28.436 [info] GET /sign_in\n","stream":"stdout","time":"2024-03-20T22:41:28.436804982Z"}
> 
> {"log":"2024-03-20 22:41:28.438 [info] Sent 200 in 1ms\n","stream":"stdout","time":"2024-03-20T22:41:28.438637109Z"}
> 
> {"log":"2024-03-20 22:41:45.136 [error] POST https://auth.tesla.com/oauth2/v3/token -\u003e error: \"non-existing domain\" (8001.655 ms)\n","stream":"stdout","time":"2024-03-20T22:41:45.`
> 
> _Originally posted by @LadPack in https://github.com/teslamate-org/teslamate/discussions/3213#discussioncomment-8858750_

**Comment by LadPack** [comment]
I would just like to add, I am the person having the above issue as well.

Mine initially presented exactly the same way, car credentials were saved, but car showed unavailable, then trying to re enter my credentials from the official tool gave invalid token errors.

**Comment by lemondjeff** [comment]
I just started seeing this as well after upgrading to the 1.28.5 Docker image. I downgraded to 1.28.4, which was working previously, and still had the same issue, so I'm wondering if this is really an issue with TeslaMate or if something else has changed on Tesla's end.

**Comment by lemondjeff** [comment]
I think something got borked with DNS on my QNAP NAS (which hosts the TeslaMate Docker image). I rebooted the NAS, and now TeslaMate is resolving domains and working again.

**Comment by lits101** [comment]
I've tried rebooting the docker host machine and this didn't resolve my issue.

**Comment by cwanja** [comment]
@LadPack and @lits101 - where are your docker installations hosted?

**Comment by LadPack** [comment]
My docker installation is hosted on a proxmox lxc running Debian.

Yesterday I tried changing my DNS from 8.8.8.8 to 1.1.1.1 and that did not
work. Also tried a few different reboots including the whole server

**Comment by lits101** [comment]
On a raspberry pi on my local network. Sent from my iPhoneOn 26 Mar 2024, at 10:57 am, Chris Wanja ***@***.***> wrote:﻿
@LadPack and @lits101 - where are your docker installations hosted?

**Comment by cwanja** [comment]
@lits101 Do you have any custom network configuration in your Docker compose file? Is the Tesla app on the same network able to work? What is your DNS provider? 

This seems local to your installation - other installs are working on the latest version and not reporting issues. Thus, I am moving this to a discussion and not an issue.

**Reply by lits101** [reply]
There's nothing special about my setup. 

I'm running whatever docker_compose.yaml example the docs says to use. The Tesla app is on the same network and works fine. I'm using my internet service provider's DNS.

**Reply by cwanja** [reply]
Did you reset your Tesla password at any point? What happens if you try and provide new tokens in TeslaMate 1.25.4? 

Last resort would be to backup, blow away TeslaMate, reinstall and restore from back. But it may not resolve the issue if your host machine / local network is blocking things.

You mentioned [here](https://github.com/teslamate-org/teslamate/discussions/3213#discussioncomment-8900394) that you could not access api.github.com and x.com - which leads towards a fundamental network issue on your host machine unrelated to TeslaMate.

**Reply by lits101** [reply]
No password reset. 

Yeah it seems like a network issue for sure. But I can ping those domains from the docker host just fine, and elsewhere on my network. I'm using the vanilla docker-compose.yaml from [here](https://docs.teslamate.org/docs/installation/docker), sans the config for mosquitto which I'm hosting elsewhere. 

I'm not sure where it would be picking up dud DNS settings from.

**Reply by lits101** [reply]
I've just created new tokens, which don't work. I'm still getting those DNS issues inside the container.

**Reply by lits101** [reply]
It's definitely got something to do with DNS on my docker host. I had an old version of Teslamate on another host which I'd spun down. After spinning it back up, it appears to be working perfectly.

**Reply by cwanja** [reply]
> It's definitely got something to do with DNS on my docker host. I had an old version of Teslamate on another host which I'd spun down. After spinning it back up, it appears to be working perfectly. 

At least you have a working version. Not sure blowing away the Docker installation and volumes is going to solve the issue. But at this point, might be a good option.

**Reply by lits101** [reply]
Dumping the volumes didn't solve the issue either. I did a full update of Docker too, just in case. I'm still getting that error on the primary host. 

In my haste, I forgot to do a database backup prior to running 'docker compose down -v', so I just lost a few months of data 😂

Oh well...

**Comment by Str1atum** [comment]
Same problem since today, didn't chance anything:

2024-03-27 15:28:01.093 [warning] Using fallback vehicles:
2024-03-27 15:28:01.088 [warning] Could not get vehicles: :unknown
2024-03-27 15:28:01.088 [warning] TeslaApi.Error / non-existing domain
2024-03-27 15:28:01.087 [error] GET https://owner-api.teslamotors.com/api/1/products -> error: "non-existing domain" (8002.203 ms)
[
  %TeslaApi.Vehicle{
    id: xxx,
    vehicle_id: xxx,
    vin: "xxx",
    tokens: [],
    state: "unknown",
    option_codes: [],
    in_service: false,
    display_name: "xxx",
    color: nil,
    calendar_enabled: nil,
    backseat_token: nil,
    backseat_token_updated_at: nil,
    api_version: nil,
    charge_state: nil,
    climate_state: nil,
    drive_state: nil,
    gui_settings: nil,
    vehicle_config: nil,
    vehicle_state: nil
  }
]
2024-03-27 15:28:01.093 [info] Starting logger for 'xxx'
2024-03-27 15:28:01.108 [info] MQTT connection has been established
2024-03-27 15:28:09.106 [error] GET https://owner-api.teslamotors.com/api/1/vehicles/XXXX -> error: "non-existing domain" (8002.448 ms)
2024-03-27 15:28:09.106 [warning] TeslaApi.Error / non-existing domain
2024-03-27 15:28:09.106 car_id=2 [error] Error / :unknown

**Comment by roughwood** [comment]
Getting this as well
![image](https://github.com/teslamate-org/teslamate/assets/7537179/9d216315-7636-4d85-8bd9-a7586b4a641c)
![image](https://github.com/teslamate-org/teslamate/assets/7537179/e95c66a7-b3ab-4550-a0bb-c40a13f1055c)

Rebooting the container several times seems to fix it, but I'm experiencing this failure several times per day. It never seems to recover on its own.

**Reply by lemondjeff** [reply]
As noted above (and this was true for me as well) you may need to reboot the Docker **host**, not just the container itself. That seemed to fix my DNS issues.
